### DISCUSSION TITLE
Tags, Properties, and Categories

### DISCUSSION THREAD

**Discussion by frozendnn** [DNNCommunity/Dnn.CommunityForums#1430] [Q&A]
How are they different and why are they different? I feel like they are redundant and probably makes the code complicated (like I know). It definently confuses me as to why and how to use them.

So inside a forum a topic can be divided into (predefined) categories (like tags?)
and a topic can have tags (predefined or user genertaed), 
but what are Properties then? 

Maybe they all are great, but how can all three used for one topic in one forum? 

Thanks!

**Comment by johnhenley** [comment]
I will agree that they are all confusing

### Tags
Tags categorize posts and allow users to search topics more easily. The latest releases of the Forums module include tags as part of the DNN search integration. Additionally, tags can be clicked on to show all posts with that tag. Users can enter/create their own tags by typing them in when creating the post, so there is no admin "cost" other than setting up security to allow for tags. As tags are entered by users, they become part of the forums pool of tags, and can be reused by others, so it's a way of building a loose taxonomy. I think I created an issue/feature request where it would be nice to integrate this directly with the DNN taxonomy APIs. This is from some very old AF documentation:
![image](https://user-images.githubusercontent.com/9553126/213466164-593aa89e-efc5-49e6-9913-903787d6cf37.png)

### Categories
Categories are ways of "filing" posts into buckets, such like if you had a forum set up to support a helpdesk, and wanted to organize the posts into different subjects. (Personally I've never used this). From old AF documentation:
<img width="997" height="792" alt="image" src="https://github.com/user-attachments/assets/41f64381-37b1-41f9-9730-c75d372b6e14" />

### Properties
Properties are yet another way of categorizing posts, but in a more structured way if desired. A property of a topic can be a yes/no, freeform text, or single or multiple selections from a DNN list. So you can actually build a pretty robust solution if you wanted to capture a lot of information on a post. Thinking about a helpdesk again, you could capture machine name (freeform text), Windows vs. Mac (single selection from a list), etc.

**Comment by WillStrohl** [comment]
@johnhenley Am I mis-remembering in thinking that you can also use `Properties` for integrations as well?

**Comment by WillStrohl** [comment]
@frozendnn It's not at all a big deal, but this maybe would have been better as a [Discussion](https://github.com/DNNCommunity/Dnn.CommunityForums/discussions).  We'll answer you either way, but that has better tooling for Q&A things. Thank you again for using the forums and chatting with us either way.  🙂

**Comment by johnhenley** [comment]
@WillStrohl There is a feature in github to convert issues (based on label) into discussion items. So you could convert all "question" issues into discussion items.

[https://docs.github.com/en/discussions/managing-discussions-for-your-community/managing-discussions#converting-issues-based-on-labels](url)

![image](https://user-images.githubusercontent.com/9553126/218152488-04d4afee-2167-4366-b5a7-64927698a689.png)

**Comment by johnhenley** [comment]
> How are they different and why are they different? I feel like they are redundant and probably makes the code complicated (like I know). It definently confuses me as to why and how to use them.
> 
> 
> 
> So inside a forum a topic can be divided into (predefined) categories (like tags?)
> 
> and a topic can have tags (predefined or user genertaed), 
> 
> but what are Properties then? 
> 
> 
> 
> Maybe they all are great, but how can all three used for one topic in one forum? 
> 
> 
> 
> Thanks!
> 
> 

@frozendnn are you still using forums?
